docs/standard/native-interop/best-practices.md

```diff
@@ -0,0 +1,245 @@
+| 32    | `LONG`           | `long`               | `int`    |                                      |
+| 32    | `ULONG`          | `unsigned long`      | `uint`   |                                      |
+| 32    | `DWORD`          | `unsigned long`      | `uint`   |                                      |
+| 64    | `QWORD`          | `long long`          | `long`   |                                      |
+| 64    | `LARGE_INTEGER`  | `long long`          | `long`   |                                      |
+| 64    | `LONGLONG`       | `long long`          | `long`   |                                      |
+| 64    | `ULONGLONG`      | `unsigned long long` | `ulong`  |                                      |
+| 64    | `ULARGE_INTEGER` | `unsigned long long` | `ulong`  |                                      |
+| 32    | `HRESULT`        | `long`               | `int`    |                                      |
+| 32    | `NTSTATUS`       | `long`               | `int`    |                                      |
+
+
+The following types, being pointers, do follow the width of the platform. Use `IntPtr`/`UIntPtr` for these.
+
+| Signed Pointer Types (use `IntPtr`) | Unsigned Pointer Types (use `UIntPtr`) |
+|:------------------------------------|:---------------------------------------|
+| `HANDLE`                            | `WPARAM`                               |
+| `HWND`                              | `UINT_PTR`                             |
+| `HINSTANCE`                         | `ULONG_PTR`                            |
+| `LPARAM`                            | `SIZE_T`                               |
+| `LRESULT`                           |                                        |
+| `LONG_PTR`                          |                                        |
+| `INT_PTR`                           |                                        |
+
+A Windows `PVOID` which is a C `void*` can be marshaled as either `IntPtr` or `UIntPtr`, but prefer `void*` when possible.
+
+[Windows Data Types](/windows/desktop/WinProg/windows-data-types)
+
+[Data Type Ranges](/cpp/cpp/data-type-ranges)
+
+## Structs
+
+Managed structs are created on the stack and aren't removed until the method returns. By definition then, they are "pinned" (it won't get moved by the GC). You can also simply take the address in unsafe code blocks if native code won't use the pointer past the end of the current method.
+
+Blittable structs are much more performant as they can simply be used directly by the marshalling layer. Try to make structs blittable (for example, avoid `bool`). For more information, see the [Blittable Types](#blittable-types) section.
+
+*If* the struct is blittable, . As mentioned above, you can validate that the type is blittable by attempting to create a pinned `GCHandle`. If the type is not a string or considered blittable, `GCHandle.Alloc` will throw an `ArgumentException`.
```

Comment:
Last feedback and I think we're good to merge. This sentence is still unfinished.